src/server/plugin/AuthCore.ts

```diff
@@ -33,9 +34,11 @@ export class AuthCore {
     return url
   }
 
-  authenticate(username: string, groups: string[]): boolean {
-    const success = groups.includes(this.requiredOrgName)
-
+  authenticate(username: string, groups: string[], teams: string[]): boolean {
+    let success = groups.includes(this.requiredOrgName)
+    if (success && this.requiredTeamName){
+      success = teams.includes(this.requiredTeamName)
+    }
```

Comment:
Groups here refers to the concept of verdaccio real_groups. That's why it's called `groups` instead of `orgs`. See `AuthCore.createAuthenticatedUser` where all plugin-specific identifiers are added to `real_groups`.

To be conceptually consistent I would prefer if the teams are included in the groups array instead of passed in as a separate array, i.e. the provider would make 2 GH calls and return the org name and team name in one array. Then we also don't need `Cache.getTeams` any more.

The same change applies to the unit tests. I'd prefer if we keep them how they currently work and the team name is just another "group" the user belongs to.

Reply:
> I would prefer if the teams are included in the groups array instead of passed in as a separate array

I'm not sure what you have in mind, if you have an example of how the `groups` array would look like with orgs and teams, it would be great

Currently the `createAuthenticatedUser` looks like :

```javascript
createAuthenticatedUser(username: string): User {
    // See https://verdaccio.org/docs/en/packages
    return {
        name: username,
        groups: ["$all", "@all", "$authenticated", "@authenticated"],
        real_groups: [username, this.requiredOrgName, this.requiredTeamName],
    }
}
```

do you mean that we would somehow concatenate `requiredOrgName` and `requiredTeamName` ?

```javascript
createAuthenticatedUser(username: string): User {
    // See https://verdaccio.org/docs/en/packages
    return {
        name: username,
        groups: ["$all", "@all", "$authenticated", "@authenticated"],
        real_groups: [username, this.requiredOrgName + "_" +  this.requiredTeamName],
    }
}
```

Also would it still work if you want to implement multiple teams filter in the future, like `dev`,`prod` and `test` teams ?

Reply:
authenticate would remain unchanged:

```ts
authenticate(username: string, groups: string[]): boolean {
```

let's say you have an org named "ORG" and a team named "TEAM", the passed-in groups argument would be

```ts
["ORG", "TEAM"]
```

Reply:
In the original source code `authenticate(username: string, groups: string[])` groups return the list of organizations the user is member of : 

https://github.com/n4bb12/verdaccio-github-oauth-ui/blob/master/src/server/plugin/AuthCore.ts#L36-L44

so groups returns `["org1","org2","org3"]` and you check the org using `groups.includes(requiredOrg)`

In my addition, I've added trhe teams array which gives you a list of teams the user is member of like `
["team1","team2"]`